### PR TITLE
build(local): Develop natively on the local host

### PR DESCRIPTION
**Pull request by rpatterson**
Adds a `./local-devel/Makefile` and some `./local-devel/bin/*` scripts that can be used
to stitch together, via symbolic links, the buildout configuration files from VCS
checkouts of the various repositories that are used to build the Docker image and their
base layers.  This way the buildout can be run locally against a configuration that
should be the same as the one used in the built images and changes made to that
configuration should be visible in `$ git status` in the checkouts they are linked from.

----

All of the changes here have been made to be isolated to the `./local-devel/`
subdirectory and should not affect anything or anyone else.  Since I'm now dependent on
these changes for my development, the tax for keeping this branch up to date will only
increase over time, so I'm hoping I can get this PR merged even if it's not widely used.
The only commits that touch anything outside of the `./local-devel/` subdirectory are
general improvements and should be pretty safe:

- [build(relstorage): Support alternate Postgres
  port](https://github.com/eea/eea.docker.plone/commit/fa3c70390538767ddd3118a8b005c7faebf1ee8c)
- [build(local): Capture SSH gateway/tunnel
  details](https://github.com/eea/eea.docker.plone/commit/fa3c70390538767ddd3118a8b005c7faebf1ee8c)
- [build(local): Switch Rancher CLI to rsync
  service](https://github.com/eea/eea.docker.plone/commit/a1d16973fae3990b2896b7327ca24f8c48612aa9)
- [build(local): Local devel shouldn't break image](https://github.com/eea/eea.docker.plone/commit/d6e82c7b7ae3a3204eb32232038c55c53a26ffb4)
- [build(local): Ensure same host OS pkgs installed](https://github.com/eea/eea.docker.plone/commit/3437b0d1617886a358a432c71421ef953ee40096)

So far, this has only been tested and used on an Ubuntu Groovy 20.10 host that was
largely already set up for local development.  As such there is likely more work needed
to support other development hosts (e.g. OS X).  There are also likely clean build
issues that will be discovered even on other Ubuntu hosts that don't have or have
different packages than my host.  Please test and report issues.

### CODE REVIEW

File: local-devel/Makefile [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+# Defaults if not defined in `./.env`
+COMPOSE_FILE=./docker-compose.local.yml:../docker-compose.yml
+RANCHER_STACK_PREFIX=$(USER)
+PYTHON_EXECUTABLE=python3
+ZC_BUILDOUT_VERSION=2.13.4
+SETUPTOOLS_VERSION=51.3.3
+WHEEL_VERSION=0.36.2
+# Version from `FROM plone:...` in `../Dockerfile`
+PLONE_VERSION=5.2.4
```
Comment (avoinea) [inline local-devel/Makefile:21]: Within next iteration I think we could auto-extract this versions from Dockerfile

File: local-devel/base.cfg [hunk 2]
```diff
@@ -0,0 +1 @@
+./plone/base.cfg
```
Comment (avoinea) [inline local-devel/base.cfg:1]: I'm not sure I've got this one. Do I have to manually `git clone` something here?

File: local-devel/bin/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+!/rancher-rsh
+!/mv-backup
```
Comment (avoinea) [inline local-devel/bin/.gitignore]: Nitpicking: Newline :smile:

File: src/docker/docker-setup.sh [hunk 4]
```diff
@@ -1,50 +1,9 @@
...
-  lynx
-  poppler-utils
-  subversion
-  tex-gyre
-  vim
-  wv
-"
+buildDeps=$(cat "/build-deps.txt")
+
+runDeps=$(cat "/run-deps.txt")
```
Comment (avoinea) [inline src/docker/docker-setup.sh:6]: Nice touch! :+1:

**Review by avoinea — APPROVED** [review]

File: local-devel/Makefile (hunk 1 above)
Comment (rpatterson) [reply local-devel/Makefile:21]: Yeah, I would love to do that.  I felt I was pushing it a bit in terms of amount of time spent vs degree of automation so I just copied and pasted these.

File: local-devel/base.cfg (hunk 2 above)
Comment (rpatterson) [reply local-devel/base.cfg:1]: We [list the buildout configuration files from the base Plone image](https://github.com/eea/eea.docker.plone/blob/build-local-devel/local-devel/Makefile#L49) then we [extract those files into that `./local-devel/plone/` directory](https://github.com/eea/eea.docker.plone/blob/build-local-devel/local-devel/Makefile#L162https://github.com/eea/eea.docker.plone/blob/build-local-devel/local-devel/Makefile#L162) and finally we [symlink them to files like these as appropriate](https://github.com/eea/eea.docker.plone/blob/build-local-devel/local-devel/Makefile#L152).  Did you try running `$ make` and the target of that link still wasn't in place?

File: local-devel/bin/.gitignore (hunk 3 above)
Comment (rpatterson) [reply local-devel/bin/.gitignore]: /me slaps my editor's hand

Will push a fix.

File: local-devel/Makefile (hunk 1 above)
Comment (rpatterson) [reply local-devel/Makefile:21]: Actually, thinking about this further, I think The Right Solution (tm) here would be to move these into some sort of `./requirements*.txt` file in the base Plone image.  It's a more re-usable format for images that extend it in general.  Maybe I'll submit that PR upstream.

File: local-devel/base.cfg (hunk 2 above)
Comment (avoinea) [reply local-devel/base.cfg:1]: He he, "it doesn't work on my machine" :smile: 
```
sudo: apt-get: command not found
```

Comment (rpatterson) [reply local-devel/base.cfg:1]: OS X, I take it?

Comment (rpatterson) [reply local-devel/base.cfg:1]: Thanks for being my first real tester!  ;-)  I've added some more informative error message when not run on Debian/Ubuntu.  So please do test again if you have time.

Comment (avoinea) [reply local-devel/base.cfg:1]: > OS X, I take it?

ArchLinux

Comment (rpatterson) [reply local-devel/base.cfg:1]: Moved to [another PR](https://github.com/eea/eea.docker.plone/pull/5)

**Pull request merged**